mlfinlab/sampling/bootstrapping.py

```diff
@@ -6,34 +6,40 @@
 from numba import jit, prange
 
 
-def get_ind_matrix(triple_barrier_events):
+def get_ind_matrix(triple_barrier_events, price_bars):
     """
     Snippet 4.3, page 64, Build an Indicator Matrix
     Get indicator matrix
     :param triple_barrier_events: (pd.DataFrame): triple barrier events from labeling.get_events
+    :param price_bars: (pd.DataFrame): price bars which were used to form triple barrier events
     :return: (np.array) indicator binary matrix indicating what (price) bars influence the label for each observation
     """
     if bool(triple_barrier_events.isnull().values.any()) is True or bool(
             triple_barrier_events.index.isnull().any()) is True:
         raise ValueError('NaN values in triple_barrier_events, delete nans')
 
+    # Take only period covered in triple_barrier_events
+    trimmed_price_bars_index = price_bars[(price_bars.index >= triple_barrier_events.index.min()) &
+                                          (price_bars.index <= triple_barrier_events.t1.max())].index
+
     label_endtime = triple_barrier_events.t1
-    bar_index = list(triple_barrier_events.index)  # generate index for indicator matrix from t1 and index
+    bar_index = list(triple_barrier_events.index)  # Generate index for indicator matrix from t1 and index
     bar_index.extend(triple_barrier_events.t1)
-    bar_index = sorted(list(set(bar_index)))  # drop duplicates and sort
+    bar_index.extend(trimmed_price_bars_index)  # Add price bars index
+    bar_index = sorted(list(set(bar_index)))  # Drop duplicates and sort
 
     sorted_timestamps = dict(
```

Comment:
In cases like this lets put the inline comment above the code and then put the code on a single line. I think its hard to read when we have a line break in the code.